src/dune_lang/targets_spec.ml

```diff
@@ -33,43 +36,53 @@ module Kind = struct
 end
 
 module Static = struct
-  type 'path t =
-    { targets : ('path * Kind.t) list
-    ; multiplicity : Multiplicity.t
-    }
+  type 'path t = {
+  targets : ('path * Kind.t * string option) list;
+  multiplicity : Multiplicity.t;
+}
+
 end
 
 type 'a t =
   | Static of 'a Static.t
   | Infer
 
+  let named_targets : (string, String_with_vars.t) Hashtbl.t = Hashtbl.create 16
```

Comment:
Avoid introducing global mutable state. This won't work in the case where multiple different rules use the same name for targets. I expect that the name->target association can be tracked in much the same way as they are for named deps, but if that's not possible then let's discuss alternatives.